From a61209b0f5d80b4e44017078c864a275c24fbad0 Mon Sep 17 00:00:00 2001
From: ihtfp69 <ihtfp69@gmail.com>
Date: Wed, 9 Jul 2014 23:58:13 -0500
Subject: [PATCH] SystemUI: allow for the clock to be centered (2 of 2)

    This allows the clock to be centered and does
    not break the normal functionality of the status bar.
    Overflow for status bar icons is adjusted accordingly
    when the clock is centered. The clock slides in/out
    when centered and the ticker is running.

PS2: Merge status bar clock enabled and center preferences into
     a single ListPreference
PS3: Reordered the styles options

Change-Id: I4f71263d94128619e3347585e188b29ab3a7f19b
---
 res/values/cm_arrays.xml                            | 12 ++++++++++++
 res/values/cm_strings.xml                           |  5 ++++-
 res/xml/status_bar.xml                              |  9 ++++++---
 src/com/android/settings/cyanogenmod/StatusBar.java | 15 +++++++++++++++
 4 files changed, 37 insertions(+), 4 deletions(-)

diff --git a/res/values/cm_arrays.xml b/res/values/cm_arrays.xml
index b50f567..fcbe961 100644
--- a/res/values/cm_arrays.xml
+++ b/res/values/cm_arrays.xml
@@ -523,6 +523,18 @@
         <item>2</item>
     </string-array>
 
+    <string-array name="entries_status_bar_clock" translatable="false">
+        <item>@string/status_bar_clock_style_default</item>
+        <item>@string/status_bar_clock_style_center</item>
+        <item>@string/status_bar_clock_style_hidden</item>
+    </string-array>
+
+    <string-array name="values_status_bar_clock" translatable="false">
+        <item>1</item>
+        <item>2</item>
+        <item>0</item>
+    </string-array>
+
     <string-array name="entries_status_bar_battery" translatable="false">
         <item>@string/status_bar_style_icon_portrait</item>
         <item>@string/status_bar_style_icon_landscape</item>
diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 649516a..4c2afc9 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -987,7 +987,10 @@ two in order to insert additional control points. \'Remove\' deletes the selecte
 
     <!-- Status bar -->
     <string name="status_bar_title">Status bar</string>
-    <string name="status_bar_show_clock_title">Show clock</string>
+    <string name="status_bar_clock_title">Clock style</string>
+    <string name="status_bar_clock_style_default">Default</string>
+    <string name="status_bar_clock_style_center">Center</string>
+    <string name="status_bar_clock_style_hidden">Hidden</string>
     <string name="status_bar_battery_title">Battery status style</string>
     <string name="status_bar_style_icon_portrait">Icon portrait</string>
     <string name="status_bar_style_icon_landscape">Icon landscape</string>
diff --git a/res/xml/status_bar.xml b/res/xml/status_bar.xml
index ce48828..461e866 100644
--- a/res/xml/status_bar.xml
+++ b/res/xml/status_bar.xml
@@ -18,10 +18,13 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:title="@string/status_bar_title">
 
-   <com.android.settings.cyanogenmod.SystemSettingCheckBoxPreference
+    <ListPreference
         android:key="status_bar_clock"
-        android:title="@string/status_bar_show_clock_title"
-        android:defaultValue="true" />
+        android:dialogTitle="@string/status_bar_clock_title"
+        android:title="@string/status_bar_clock_title"
+        android:entries="@array/entries_status_bar_clock"
+        android:entryValues="@array/values_status_bar_clock"
+        android:defaultValue="1" />
 
     <ListPreference
         android:key="status_bar_battery"
diff --git a/src/com/android/settings/cyanogenmod/StatusBar.java b/src/com/android/settings/cyanogenmod/StatusBar.java
index f29a9d3..0734ac4 100644
--- a/src/com/android/settings/cyanogenmod/StatusBar.java
+++ b/src/com/android/settings/cyanogenmod/StatusBar.java
@@ -42,11 +42,13 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
     private static final String STATUS_BAR_SIGNAL = "status_bar_signal";
 
     private static final String STATUS_BAR_BATTERY_SHOW_PERCENT = "status_bar_battery_show_percent";
+    private static final String STATUS_BAR_CLOCK_STYLE = "status_bar_clock";
 
     private static final String STATUS_BAR_STYLE_HIDDEN = "4";
     private static final String STATUS_BAR_STYLE_TEXT = "6";
     private static final String STATUS_BAR_STYLE_STATUSBAR = "7";
 
+    private ListPreference mStatusBarClockStyle;
     private ListPreference mStatusBarBattery;
     private SystemSettingCheckBoxPreference mStatusBarBatteryShowPercent;
     private ListPreference mStatusBarCmSignal;
@@ -63,6 +65,8 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
         PreferenceScreen prefSet = getPreferenceScreen();
         ContentResolver resolver = getActivity().getContentResolver();
 
+        mStatusBarClockStyle = (ListPreference) findPreference(STATUS_BAR_CLOCK_STYLE);
+
         mStatusBarBattery = (ListPreference) findPreference(STATUS_BAR_BATTERY);
         mStatusBarBatteryShowPercent =
                 (SystemSettingCheckBoxPreference) findPreference(STATUS_BAR_BATTERY_SHOW_PERCENT);
@@ -72,6 +76,11 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
                 prefSet.findPreference(Settings.System.STATUS_BAR_BRIGHTNESS_CONTROL);
         refreshBrightnessControl();
 
+        int clockStyle = Settings.System.getInt(resolver, Settings.System.STATUS_BAR_CLOCK, 1);
+        mStatusBarClockStyle.setValue(String.valueOf(clockStyle));
+        mStatusBarClockStyle.setSummary(mStatusBarClockStyle.getEntry());
+        mStatusBarClockStyle.setOnPreferenceChangeListener(this);
+
         int batteryStyle = Settings.System.getInt(resolver, Settings.System.STATUS_BAR_BATTERY, 0);
         mStatusBarBattery.setValue(String.valueOf(batteryStyle));
         mStatusBarBattery.setSummary(mStatusBarBattery.getEntry());
@@ -133,6 +142,12 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
             Settings.System.putInt(resolver, Settings.System.STATUS_BAR_SIGNAL_TEXT, signalStyle);
             mStatusBarCmSignal.setSummary(mStatusBarCmSignal.getEntries()[index]);
             return true;
+        } else if (preference == mStatusBarClockStyle) {
+            int clockStyle = Integer.valueOf((String) newValue);
+            int index = mStatusBarClockStyle.findIndexOfValue((String) newValue);
+            Settings.System.putInt(resolver, Settings.System.STATUS_BAR_CLOCK, clockStyle);
+            mStatusBarClockStyle.setSummary(mStatusBarClockStyle.getEntries()[index]);
+            return true;
         }
 
         return false;
-- 
1.9.3

